mergify_engine/signals.py

```diff
@@ -234,6 +274,50 @@ async def send(
     ...
 
 
+@typing.overload
+async def send(
+    repository: "context.Repository",
+    pull_request: github_types.GitHubPullRequestNumber,
+    event: typing.Literal["action.queue.enter"],
+    metadata: EventQueueEnterMetadata,
+    trigger: str,
+) -> None:
+    ...
+
+
+@typing.overload
+async def send(
+    repository: "context.Repository",
+    pull_request: github_types.GitHubPullRequestNumber,
+    event: typing.Literal["action.queue.leave"],
+    metadata: EventQueueLeaveMetadata,
+    trigger: str,
+) -> None:
+    ...
+
+
+@typing.overload
+async def send(
+    repository: "context.Repository",
+    pull_request: github_types.GitHubPullRequestNumber,
+    event: typing.Literal["action.queue.checks_end"],
+    metadata: EventQueueChecksEndMetadata,
+    trigger: str,
+) -> None:
+    ...
+
+
+@typing.overload
+async def send(
```

Comment:
What is the point of all those overload ? Since every metadata specified is a child of EventMetadata and the event name can be added in the EventName literal type at the top, calling the base `send` at the bottom of this file should work without having to make all those overloads, no ?